test/types/logger.test-d.ts

```diff
@@ -5,11 +5,17 @@ import * as pino from 'pino'
 
 expectType<FastifyLoggerInstance>(fastify().log)
 
-;['trace', 'debug', 'info', 'warn', 'error', 'fatal'].forEach(logLevel => {
+class Foo {}
+
+['trace', 'debug', 'info', 'warn', 'error', 'fatal'].forEach(logLevel => {
   expectType<FastifyLogFn>(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel])
   expectType<void>(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel](''))
   expectType<void>(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel]({}))
-  expectError(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel](0))
+  expectType<void>(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel]({ foo: 'bar' }))
+  expectType<void>(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel](new Error()))
+  expectType<void>(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel](new Foo()))
+  // unknown allow number type to be passed.
+  // expectError(fastify<Server, IncomingMessage, ServerResponse, FastifyLoggerInstance>().log[logLevel as LogLevel](0))
```

Comment:
Here is the problem, should we really block `number` to be passed? If it is true, I can't think of a proper type to block it in TypeScript.
```ts
type HaveToString = { toString(): string } | Record<string | number | symbol, unknown>
```
I have tired the type to check if passed variables need to have `toString`, but `number` also match this criteria. 
@zekth @Ethan-Arrowood Any troughs on this?

Reply:
Technically, I think that `unknown` is best here and that you should remove that expectError test. I'm fairly certain it was added as a sort of _not string or object_ test and was not specific to `number` type (as in it could've been any non string or object)

Reply:
Updated